File: Duran_TuranMert.php
<?php
//Initialize
$phoneList = array("Zeynep"=>"123", "Turan"=>"3347", "Musa"=>"43");
 // Get value of Given Key
echo $phoneList['Turan'] ;
echo "\n ";

// Add a new element
$phoneList['Halil'] = "3437";
//Remove an element
unset($phoneList["Turan"]); 
//Modify the value of an existing element
$phoneList['Halil'] = "3333";
//Search for the existence of a key
if (array_key_exists("Halil",$phoneList))
  {
  echo "Key is available!\n ";
  }
else
  {
  echo "Key is not available!\n ";
  }

//Search for the existence of a value
if (in_array("43", $phoneList)) 
  { 
  echo "found\n "; 
  } 
else
  { 
  echo "not found\n "; 
  } 
function foo($list) {
// Print key-value pair with function
  foreach($list as $key => $value) {
    echo "Key=" . $key . ", Value=" . $value;
    echo "\n"; 
  }
}
foo($phoneList);

//https://www.w3schools.com/php/php_arrays_associative.asp
//https://www.tutorialrepublic.com/faq/how-to-delete-an-element-from-an-array-in-php.php
?> 

File: Duran_TuranMert.rs
use std::collections::HashMap;

fn main() {
    // Initializing 
    let mut phoneList = HashMap::new();
    phoneList.insert(String::from("Turan"), 13242);
    phoneList.insert(String::from("Ahmet"), 3454);
    phoneList.insert(String::from("Radman"), 2423);
    phoneList.insert(String::from("Gunes"), 430);
    // Getting Value with key
    println!("{}", phoneList["Turan"]);
    // Add a new element
    phoneList.insert(String::from("Zeynep"), 431);
    // Remove an element
    phoneList.remove("Turan");
    // Modify value
    *phoneList.get_mut("Gunes").unwrap() = 432;
    println!("{}", phoneList["Gunes"]);
    //Search for the existence of a key
    println!("{}", phoneList.contains_key("Gunes"));
    //Search for the existence of a value
    let mut searchedValue : i32;
    searchedValue = 2423;
    checkValue(&mut phoneList, &mut searchedValue);
    //println!("{}", phoneList.containsValue(430));
    printAll(&mut phoneList);

}
    fn printAll(phoneList: &mut HashMap<String, i32> ){
        for (key, value) in phoneList.iter() {
        println!("{} - {}", key, value);
      }
    }
    fn checkValue(phoneList: &mut HashMap<String, i32>, no: &mut i32){
        for (key, value) in phoneList.iter() {
        if no == value{
          println!("Value found in List!");
        }
      }
    }